instrumentation/spring/spring-kafka-2.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/kafka/v2_7/SpringKafkaTest.java

```diff
@@ -358,44 +407,79 @@ void shouldHandleFailureInKafkaBatchListener() {
                               stringAssert -> stringAssert.startsWith("producer"))));
 
           producer.set(trace.getSpan(1));
-        },
-        trace ->
-            trace.hasSpansSatisfyingExactly(
-                span ->
-                    span.hasName("testBatchTopic receive")
-                        .hasKind(SpanKind.CONSUMER)
-                        .hasNoParent()
-                        .hasAttributesSatisfyingExactly(
-                            equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
-                            equalTo(
-                                SemanticAttributes.MESSAGING_DESTINATION_NAME, "testBatchTopic"),
-                            equalTo(SemanticAttributes.MESSAGING_OPERATION, "receive"),
-                            equalTo(
-                                SemanticAttributes.MESSAGING_KAFKA_CONSUMER_GROUP,
-                                "testBatchListener"),
-                            satisfies(
-                                SemanticAttributes.MESSAGING_CLIENT_ID,
-                                stringAssert -> stringAssert.startsWith("consumer")),
-                            equalTo(SemanticAttributes.MESSAGING_BATCH_MESSAGE_COUNT, 1)),
-                span ->
-                    span.hasName("testBatchTopic process")
-                        .hasKind(SpanKind.CONSUMER)
-                        .hasParent(trace.getSpan(0))
-                        .hasLinks(LinkData.create(producer.get().getSpanContext()))
-                        .hasStatus(StatusData.error())
-                        .hasException(new IllegalArgumentException("boom"))
-                        .hasAttributesSatisfyingExactly(
-                            equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
-                            equalTo(
-                                SemanticAttributes.MESSAGING_DESTINATION_NAME, "testBatchTopic"),
-                            equalTo(SemanticAttributes.MESSAGING_OPERATION, "process"),
-                            equalTo(
-                                SemanticAttributes.MESSAGING_KAFKA_CONSUMER_GROUP,
-                                "testBatchListener"),
-                            satisfies(
-                                SemanticAttributes.MESSAGING_CLIENT_ID,
-                                stringAssert -> stringAssert.startsWith("consumer")),
-                            equalTo(SemanticAttributes.MESSAGING_BATCH_MESSAGE_COUNT, 1)),
-                span -> span.hasName("consumer").hasParent(trace.getSpan(1))));
+        });
+
+    if (Boolean.getBoolean("testLatestDeps")) {
+      // latest dep tests call receive once and only retry the failed process step
+      assertions.add(
```

Comment:
this test is a bit hard to follow - maybe extract some common assertions

Reply:
extracted common assertions

Reply:
great - much easier to read :smile: